crypto/src/e2e_identity/state.rs

```diff
@@ -0,0 +1,268 @@
+use crate::{
+    mls::credential::ext::CredentialExt,
+    prelude::{ConversationId, CryptoResult, MlsCentral, MlsConversation},
+};
+use wire_e2e_identity::prelude::WireIdentityReader;
+
+/// Indicates the state of a Conversation regarding end-to-end identity.
+/// Note: this does not check pending state (pending commit, pending proposals) so it does not
+/// consider members about to be added/removed
+#[derive(Debug, Clone, Copy, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[repr(u8)]
+pub enum E2eiConversationState {
+    /// All clients have a valid E2EI certificate
+    Valid,
+    /// Some clients are either still Basic or their certificate is expired
+    Degraded,
+    /// All clients are still Basic. If all client have expired certificates, [E2eiConversationState::Degraded] is returned.
+    Invalid,
+}
```

Comment:
The naming might be confusing I think
Maybe something like `Verified|Degraded|NotEnabled` might be better?